### PR TITLE
Improve types.rs

### PR DESCRIPTION
**Pull request by Yamakaky**
Please review my comments and propose your owns when they are missing.

---
This change is [Reviewable](https://reviewable.io/reviews/crabtw/rust-bindgen/295)

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -9,15 +9,21 @@ pub use self::Type::*;
...
+    /// A C composed type, like a struct or union.
     GComp(Rc<RefCell<CompInfo>>),
     GCompDecl(Rc<RefCell<CompInfo>>),
+    /// A C enum.
     GEnum(Rc<RefCell<EnumInfo>>),
     GEnumDecl(Rc<RefCell<EnumInfo>>),
     GVar(Rc<RefCell<VarInfo>>),
+    /// A function prototype, like `int func();`
     GFunc(Rc<RefCell<VarInfo>>),
+    /// Something
```
Comment (nox) [inline src/types.rs]: "Something else."

File: src/types.rs [hunk 2]
```diff
@@ -69,30 +75,45 @@ impl fmt::Debug for Global {
...
 pub struct FuncSig {
+    /// The return type of the function.
     pub ret_ty: Box<Type>,
+    /// The arguments of the function.
     pub args: Vec<(String, Type)>,
+    /// Is the function variadic?
     pub is_variadic: bool,
+    /// Must it be called in an `unsafe` block?
     pub is_safe: bool,
+    /// What is it's ABI?
```
Comment (nox) [inline src/types.rs]: "The ABI of the function."

File: src/types.rs [hunk 3]
```diff
@@ -9,15 +9,21 @@ pub use self::Type::*;
...
 pub enum Global {
+    /// A type definition, like `typedef int my_type;`.
     GType(Rc<RefCell<TypeInfo>>),
+    /// A C composed type, like a struct or union.
     GComp(Rc<RefCell<CompInfo>>),
     GCompDecl(Rc<RefCell<CompInfo>>),
+    /// A C enum.
     GEnum(Rc<RefCell<EnumInfo>>),
     GEnumDecl(Rc<RefCell<EnumInfo>>),
     GVar(Rc<RefCell<VarInfo>>),
```
Comment (nox) [inline src/types.rs]: "A C global variable."

File: src/types.rs [hunk 4]
```diff
@@ -124,6 +146,7 @@ impl Type {
         }
     }
 
+    /// Does the type can have the attribute `#[derive(Debugj)]`?
```
Comment (nox) [inline src/types.rs]: "Whether Debug can be derived for the type."

**Comment by Yamakaky** [comment]
Should be good. Squash + merge?

**Pull request merged**